File: modules/ShizukuSceneFix/app/src/main/java/com/example/shizukulistfix/utils/LogX.java
package com.example.shizukulistfix.utils;

import android.util.Log;

public class LogX {

    public static final String TAG = "ShizukuListFix";
    private static boolean sDebug = true;

    private LogX() {
    }

    public static void setDebug(boolean debug) {
        sDebug = debug;
    }

    public static boolean isDebug() {
        return sDebug;
    }

    public static void i(String msg) {
        Log.i(TAG, msg);
    }

    public static void d(String msg) {
        if (sDebug) {
            Log.d(TAG, msg);
        }
    }

    public static void w(String msg) {
        Log.w(TAG, msg);
    }

    public static void e(String msg) {
        Log.e(TAG, msg);
    }

    public static void e(String msg, Throwable t) {
        Log.e(TAG, msg, t);
    }

    public static void d(String format, Object... args) {
        if (sDebug) {
            Log.d(TAG, String.format(format, args));
        }
    }

    public static void i(String format, Object... args) {
        Log.i(TAG, String.format(format, args));
    }

    public static void e(String format, Object... args) {
        Log.e(TAG, String.format(format, args));
    }
}

File: modules/AdBlockerX_NoRoot/app/src/main/java/com/adblockerx/noroot/utils/LogX.kt
package com.adblockerx.noroot.utils

import android.util.Log

/**
 * 统一日志工具
 * TAG: AdBlockerX，可用 adb logcat -s AdBlockerX:V 查看
 */
object LogX {
    private const val TAG = "AdBlockerX"
    var debugEnabled = true

    fun d(msg: String) { if (debugEnabled) Log.d(TAG, msg) }
    fun i(msg: String) { Log.i(TAG, msg) }
    fun w(msg: String) { Log.w(TAG, msg) }
    fun e(msg: String) { Log.e(TAG, msg) }
    fun e(msg: String, t: Throwable) { Log.e(TAG, msg, t) }
}
